File: specification/specs2/cluster/nodes_hot_threads/NodesHotThreadsRequest.ts
@rest_spec_name("nodes.hot_threads")
class NodesHotThreadsRequest extends RequestBase {
	@request_parameter()
	interval: Time;
	@request_parameter()
	snapshots: long;
	@request_parameter()
	threads: long;
	@request_parameter()
	ignore_idle_threads: boolean;
	@request_parameter()
	thread_type: ThreadType;
	@request_parameter()
	timeout: Time;
}

File: output/csharp/Cluster/NodesHotThreads/NodesHotThreadsRequest.cs
using Nest.CommonOptions;
using Nest.Internal;
using Nest.Common;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cluster {

	public class NodesHotThreadsRequest : RequestBase {
		
		[DataMember(Name="ignore_idle_threads")]
		public bool IgnoreIdleThreads { get; set; }


		[DataMember(Name="interval")]
		public Time Interval { get; set; }


		[DataMember(Name="snapshots")]
		public long Snapshots { get; set; }


		[DataMember(Name="threads")]
		public long Threads { get; set; }


		[DataMember(Name="thread_type")]
		public ThreadType ThreadType { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }

	}
}
